core/arch/x86/mangle.c

```diff
@@ -2391,6 +2392,21 @@ mangle_interrupt(dcontext_t *dcontext, instrlist_t *ilist, instr_t *instr,
 }
 
 /***************************************************************************
+ * Single step exceptions generation
+ */
+void
+mangle_single_step(dcontext_t *dcontext, instrlist_t *ilist, instr_t *instr)
+{
+    /*
+     * Simply inserts a nop so as that next instruction where a single step
+     * exception might occur is not in a different basic block.
+     * FIXME i#2144 : to be absolutely transparent, we should translate the
+     * exception address as if we did not insert this nop.
+     */
+    POST(ilist, instr, INSTR_CREATE_nop(dcontext));
```

Comment:
Hmm, the instrlist_set_translation_target() is probably automagically giving this a translation target but it's equal to the popf, which is not what we want, right?  Also, when translating the single-step exception that points at the nop, won't  translate_walk_track() consider this "unsupported_mangle" and thus translate_walk_good_state() will return false and we'll get an assert?

So we need two things: one, set the translation to the next app pc; and two, recognize the nop during translation.